echidna/core/scale.py

```diff
@@ -58,25 +58,33 @@ def scale(self, spectrum, dimension, **kwargs):
         axis = spectrum.get_config().get_index(dimension)
         par = spectrum.get_config().get_par(dimension)
         low = par._low
+        high = par._high
         n_bins = par._bins
         step = par.get_width()
         for bin in range(n_bins):
             x = par.get_bin_centre(bin)
-            if x/sf < low or x/sf > par._high:
+            if x/sf < low or x/sf > high:
                 continue  # Trying to scale values outside range (Unknown)
+            old_bin1 = par.get_bin(x/sf)
+            old_bin_centre1 = par.get_bin_centre(old_bin1)
             y = interpolation(x/sf)
             if y <= 0.:
                 continue
-            old_bin1 = par.get_bin(x/sf)
-            old_bin_centre1 = par.get_bin_centre(old_bin1)
+            if numpy.isnan(y):
+                y = spectrum.project(dimension)[old_bin1]
             if par.get_bin_centre(old_bin1) > x/sf:
                 old_bin2 = old_bin1 - 1
                 if old_bin2 >= 0:
                     x_low1 = old_bin_centre1 - 0.5*step  # Equals x_high2
                     x_high1 = x/sf + 0.5*step
-                    area1 = numpy.fabs(0.5 * (x_high1 - x_low1) *
-                                       (interpolation(x_high1) +
-                                        interpolation(x_low1)))
+                    if x_high1 > high - 0.5*step:
+                        # Assume uniform distribution in last bin
+                        area1 = (x_high1 - x_low1) * \
+                            spectrum.project(dimension)[-1]
+                    else:
+                        area1 = numpy.fabs(0.5 * (x_high1 - x_low1) *
+                                           (interpolation(x_high1) +
+                                            interpolation(x_low1)))
```

Comment:
However the problem then comes to this line. I you were to make the modification above, `area1` here evaluates as a very small number ~10^-13. This is because you are now trying to integrate almost the entire spectrum using on trapezium.

This raises the issue again as to why we go to all the trouble of calculating the interpolation function, only to then make an approximation for the integration. Admittedly if you have fine enough bins, this approximation is small, but I think if you use any thing more accurate than a linear interpolation (quadratic, cubic etc.) the increase in accuracy will be negated by this approximation.